client/asset/eth/eth.go

```diff
@@ -300,8 +303,18 @@ func (eth *ExchangeWallet) Balance() (*asset.Balance, error) {
 	return bal, nil
 }
 
-// getInitGas gets an estimate for the gas required for initiating a swap.
+// getInitGas gets an estimate for the gas required for initiating a swap. The
+// result is cached and reused in subsequent requests.
+//
+// This function will return an error if the wallet has a non zero balance.
```

Comment:
Will it error if wallet has a zero balance or non-zero? Does it need at least the init gas fee?

Reply:
The reason it errors is because initializing a swap requires you to send a non zero value. It doesn't error due to a gas fee. I set the gas price to 0 in `estimateGas`.